superset-frontend/src/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -145,13 +146,38 @@ export default class SqlEditorLeftBar extends React.PureComponent {
         <div className="divider" />
         <StyledScrollbarContainer>
           <StyledScrollbarContent contentHeight={tableMetaDataHeight}>
-            {this.props.tables.map(table => (
-              <TableElement
-                table={table}
-                key={table.id}
-                actions={this.props.actions}
-              />
-            ))}
+            <Collapse
+              ghost
+              expandIconPosition="right"
+              css={theme => css`
```

Comment:
I saw in other components in superset project we use `styled-components` style to build css like:
![image](https://user-images.githubusercontent.com/56388545/106762905-fc30e380-663e-11eb-92b0-d443728255cc.png)

so I think it can be good all other new components design with same patterns :)

Reply:
Great question! I'm using the css prop instead of styled components as you can see in [Button](https://github.com/apache/superset/pull/12832) PR. Actually @rusackas asked me why I prefer this style on that PR and this was my answer:

For me the use of a property seems more natural to the objective of customizing a component. [Here](https://benmcmahen.com/why-i-choose-emotion/#:~:text=The%20styled%20method%20uses%20the,to%20create%20buttons%20with%20styles.&text=With%20examples%20like%20this%2C%20it's,d%20use%20the%20style%20prop.) you will find a detailed explanation but to summarize:

- You’re still writing regular React components
- Object styles are easier to work with
- Naming things is hard
- You colocate styles with elements
- Composition is dead easy

I'm totally open for different opinions. @rusackas what do you think?

Reply:
I'm kind of on the fence. Both are kind of nice. The `css` prop is good for all the reasons you mention. The `styled.x` approach is also nice because it's kind of like having a CSS file , and you can put multiples of them in a single file in one place. Unless there's a performance difference (or one makes the stupid sourcemaps actually _work_) then I'm pretty ambivalent, so long as we use the Theme vars wherever we can, and kill off any legacy LESS we no longer need.

Reply:
I'm a bigger fan of the `css` prop only because creating a new component is maintenance than just dumping things inline. Unless we will reuse the style in multiple components or the CSS is really really long, there is no additional benefit of creating new styled components.

On a related note, I think it'd also be easier to use more nested styles than creating separate styled components for each child element (like [here](https://github.com/apache/superset/blob/465d986617233ab401c99f9eab0f55488e0dae09/superset-frontend/src/dashboard/components/nativeFilters/FilterBar.tsx#L122) and [here](https://github.com/apache/superset/blob/c72c39bffd2ac881629b35b906fe6400c4bfe5f2/superset-frontend/src/dashboard/components/nativeFilters/CancelConfirmationAlert.tsx#L25))

You may have a higher change of css classname conflict, but that should be manageable with proper namespacing.

Reply:
I know that this discussion already closed, but anyway I'd like still ask about something that looks strange for me:
```You colocate styles with elements```
Actually when you have some more complex component not like simple button, when you start mix styles with JSX/JS logic you miss JSX structure of component because styles can take a lot of place, and also some js logic can be missed from eye

Reply:
@simcha90 When styles get bigger, that's generally an indication that your component should be broken into smaller components. If that's not the case, if your component is at the atomic level but has a lot of different CSS states then what I usually do is extract the `css` prop to a variable next to the component declaration. That way the style is still co-located with the element and we preserve the benefits listed above.